src/pages/multi-signup.js

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect } from "react"
+import { useForm } from "react-hook-form"
+import "../styles/signup.css"
+import events from '../../dist/events.json'
+
+export default function App({ location }) {
+    const [matchingEvents, setMatchingEvents] = useState([])
+    const [submissionStatus, setSubmissionStatus] = useState({})
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+    } = useForm()
+
+    useEffect(() => {
+        const params = new URLSearchParams(location.search)
+        const searchTerm = params.get("search")
+        if (searchTerm) {
+            const now = new Date()
+
+            // First filter by search term and future dates
+            const filteredEvents = events.filter(event => {
+                const eventStart = new Date(event.start)
+                return event.title.toLowerCase().includes(searchTerm.toLowerCase()) &&
+                    eventStart > now
+            })
+
+            // Then filter out duplicate events by title
+            const seenTitles = new Set()
+            const uniqueEvents = filteredEvents.filter(event => {
+                if (seenTitles.has(event.title)) {
+                    return false
+                }
+                seenTitles.add(event.title)
+                return true
+            })
+
+            setMatchingEvents(uniqueEvents)
+            // Initialize submission status for each event
+            const initialStatus = {}
+            uniqueEvents.forEach(event => {
+                initialStatus[event.uid] = 'pending'
+            })
+            setSubmissionStatus(initialStatus)
+        }
+    }, [location.search])
+
+    const onSubmit = async data => {
```

Comment:
Did you try wrapping this in a [`useCallback` hook](https://react.dev/reference/react/useCallback)?